cpp/src/CMakeLists.txt

```diff
@@ -107,11 +107,19 @@ target_include_directories(kim-loader PUBLIC
 target_link_libraries(kim-loader ${CMAKE_DL_LIBS})
 
 if(WIN32 AND NOT CYGWIN)
-  # Activate C++17 standard on Windows, because kim-api code uses C++ filesystem library internally.
-  set_property(TARGET kim-api PROPERTY CXX_STANDARD 17)
-  set_property(TARGET kim-api PROPERTY CXX_STANDARD_REQUIRED YES)
-  set_property(TARGET kim-loader PROPERTY CXX_STANDARD 17)
-  set_property(TARGET kim-loader PROPERTY CXX_STANDARD_REQUIRED YES)
+  # Activate C++17 language standard on Win32 platform, because kim-api code uses C++ filesystem library internally.
+  # Note: The 'cxx_std_17' target compile feature was added by CMake version 3.8. 
+  # For older CMake versions, fall back to setting the CXX_STANDARD target property instead.
+  get_property(cxx_features GLOBAL PROPERTY CMAKE_CXX_KNOWN_FEATURES) 
+  if("cxx_std_17" IN_LIST cxx_features)
+    target_compile_features(kim-api PUBLIC cxx_std_17)
```

Comment:
What about setting these only for the `KIM_FILESYSTEM_Path.cpp` file?  Would this be an advantage/disadvantage?  I'm interested in your experience/views on "best practice" in this regard....

Reply:
No, compiling only some .cpp files in C++17 mode and others in C++98 mode will likely lead to problems, because the two parts will not be ABI-compatible. In fact, I was struggling with this problem myself for quite some time last week. The ``collections-info`` executable crashed with a segmentation fault, because it was compiled in C++98 mode while the ``kim-loader`` library was compiled in C++17 mode (on Win32). At build time everything seemed to be fine (no compiler or linker warnings) but at runtime the program crashed, because ``sizeof(KIM::FILESYSTEM::Path)`` was not the same in different parts of the same executable. 
This is why I added ``target_compile_features(kim-loader PUBLIC cxx_std_17)`` in the CMake script, because it makes sure that the C++17 requirement will be passed on from the ``kim-loader`` target to other targets consuming it. This is not the case when just setting the ``CXX_STANDARD`` target property to 17.

Reply:
Nice!  Great to know. Thanks for the explanation.